src/components/ApplePayButton.tsx

```diff
@@ -0,0 +1,113 @@
+import { type StyleProp, type ViewStyle } from 'react-native';
+import { useCallback } from 'react';
+import { PlatformPayView } from '..';
+
+export const ApplePayButtonTheme = {
+  WHITE: 1,
+  WHITE_OUTLINE: 2,
+  AUTOMATIC: 3,
+  BLACK: 4,
+};
+
+export const ApplePayButtonType = {
+  BUY: 1,
+  SETUP: 2,
+  INSTORE: 3,
+  DONATE: 4,
+  CHECKOUT: 5,
+  BOOK: 6,
+  SUBSCRIBE: 7,
+  RELOAD: 8,
+  ADDMONEY: 9,
+  TOPUP: 10,
+  ORDER: 11,
+  RENT: 12,
+  SUPPORT: 13,
+  CONTRIBUTE: 14,
+  TIP: 15,
+  CONTINUE: 16,
+  PLAIN: 0,
+};
+
+export interface ApplePayButtonProps {
+  theme?: keyof typeof ApplePayButtonTheme;
+  type?: keyof typeof ApplePayButtonType;
+  radius?: number;
+  onPress?: () => void;
+  style?: StyleProp<ViewStyle>;
+}
+
+function getButtonTheme(type: string): number {
+  switch (type) {
+    case 'WHITE':
+      return 1;
+    case 'WHITE_OUTLINE':
+      return 2;
+    case 'AUTOMATIC':
+      return 3;
+    default:
+      return 0;
+  }
+}
+
+function getButtonType(type: string): number {
```

Comment:
```suggestion
function getButtonType(type: keyof typeof ApplePayButtonType;): number {
```

Reply:
Addressed in [2045d8e](https://github.com/Adyen/adyen-react-native/pull/701/commits/2045d8e83db5674ef9260ef3965040e00162ddb1)